compiler/rustc_lexer/src/cursor.rs

```diff
@@ -102,24 +111,76 @@ impl<'a> Cursor<'a> {
         Some(c)
     }
 
+    #[inline]
+    pub(crate) fn bump_if(&mut self, expected: char) -> bool {
+        let mut chars = self.chars.clone();
+        if chars.next() == Some(expected) {
+            self.chars = chars;
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Bumps the cursor if the next character is either of the two expected characters.
+    #[inline]
+    pub(crate) fn bump_if2(&mut self, expected1: char, expected2: char) -> bool {
```

Comment:
I would call this `bump_if_either`. `bump_if2` makes me think that `expected1` must be followed by `expected2`.